ns1/resource_record.go

```diff
@@ -137,12 +144,34 @@ func recordResource() *schema.Resource {
 	}
 }
 
-func genericHasher(v interface{}) int {
-	hash, err := hashstructure.Hash(v, nil)
-	if err != nil {
-		panic(fmt.Sprintf("error computing hash code for %#v: %s", v, err.Error()))
+// errJoin joins errors into a single error
+func errJoin(errs []error, sep string) error {
+	switch len(errs) {
```

Comment:
I'm unsure why these are special cases, can't these all be

```
var result bytes.Buffer
for i, err := range errs {
    if i == len(errs)-1 {
        result.WriteString(err.Error())
    } else {
        result.WriteString(err.Error())
        result.WriteString(sep)
    }
}
```
?

or am I missing something?

Reply:
from line 158:	// Remove if golang.org/issue/6714 is fixed

The vast majority of cases are likely going to be between 1-3 errors, and this code is substantially faster with less allocations than what you wrote above, if less concise.

Reply:
Ok, that's fair.  Perhaps comment that? Seems a little odd otherwise